vowpalwabbit/cb_explore_adf_cover.cc

```diff
@@ -305,17 +306,19 @@ VW::LEARNER::base_learner* setup(config::options_i& options, vw& all)
   bool with_metrics = options.was_supplied("extra_metrics");
 
   using explore_type = cb_explore_adf_base<cb_explore_adf_cover>;
-  auto data = scoped_calloc_or_throw<explore_type>(with_metrics, cover_size, psi, nounif, epsilon, epsilon_decay,
-      first_only, as_multiline(all.cost_sensitive), all.scorer, cb_type_enum, all.model_file_ver);
-
-  VW::LEARNER::learner<explore_type, multi_ex>& l = init_learner(data, base, explore_type::learn, explore_type::predict,
-      problem_multiplier, prediction_type_t::action_probs, all.get_setupfn_name(setup) + "-cover", true);
-
-  l.set_finish_example(explore_type::finish_multiline_example);
-  l.set_print_example(explore_type::print_multiline_example);
-  l.set_save_load(explore_type::save_load);
-  l.set_persist_metrics(explore_type::persist_metrics);
-  return make_base(l);
+  auto data = VW::make_unique<explore_type>(with_metrics, cover_size, psi, nounif, epsilon, epsilon_decay, first_only,
+      as_multiline(all.cost_sensitive), all.scorer, cb_type_enum, all.model_file_ver);
+  auto* l = make_reduction_learner(
+      std::move(data), base, explore_type::learn, explore_type::predict, all.get_setupfn_name(setup) + "-cover", true)
+                .set_params_per_weight(problem_multiplier)
```

Comment:
`learn_returns_prediction` should not be in the `make_reduction_learner` call. Set it with the builder instead.

```suggestion
  auto* l = make_reduction_learner(
      std::move(data), base, explore_type::learn, explore_type::predict, all.get_setupfn_name(setup) + "-cover")
                .set_learn_returns_prediction(true)
                .set_params_per_weight(problem_multiplier)
```